src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/model/CaseRecordFactory.java

```diff
@@ -0,0 +1,44 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.model;
+
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.model.caserecord.ExceptionRecord;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.model.caserecord.SupplementaryEvidenceRecord;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.model.Envelope;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.model.CaseRecord.Construct;
+import static uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.model.CaseRecord.Record;
+
+public final class CaseRecordFactory {
+
+    private static final Map<Record, Construct> AVAILABLE_MODELS = new HashMap<>();
+
+    static {
+        AVAILABLE_MODELS.put(Record.EXCEPTION_RECORD, ExceptionRecord::new);
```

Comment:
Rather do this with say an inmutable map ?
```java
ImmutableMap.of(
    EXCEPTION_RECORD, ExceptionRecord::new
    SUPPLEMENTARY_EVIDENCE,   SupplementaryEvidenceRecord::new 
)

Or even better just bake it into the enum ?
enum{
EXCEPTION_RECORD(ExceptionRecord::new); 
  private final constuct;
  Record( Construct constructor){
     construct = constructor
  } 
  public Something construct(){
     return construct()
  }
}
Unfortunately you lose the type information ... so it might be simpler to use a sealed Kotlin class for this ( its actually simpler in the long run).

Reply:
Cool. I'll go with `ImmutableEnumMap` then. In case there will be requirement of >5 records - we can alter the code